RealmSwift/MongoClient.swift

```diff
@@ -155,6 +155,21 @@ public typealias MongoUpdateBlock = (UpdateResult?, Error?) -> Void
 /// Block which returns the deleted Document on a successful delete operation, or an error should one occur.
 public typealias MongoDeleteBlock = RLMMongoDeleteBlock
 
+/// Block which returns Result.success(DocumentId) on a successful insert or Result.failure(error)
+public typealias MongoInsertResultBlock = (Result<AnyBSON?, Error>) -> Void
```

Comment:
Some of these result types should not be nullable.

Reply:
I think AnyBSON? is actually correct. BSON null is translated to `nil` and I think null is a legal document id?

Reply:
Hmm... you are correct. I still find that to be a strange use case.

Reply:
There is a possible nullable successful result in some cases. I'll check it another time as it is not for all collection methods.